lib/health_cards/health_card.rb

```diff
@@ -146,78 +169,90 @@ def to_json(*_args)
     # @return [Hash] A hash with the same content as the FHIR::Bundle, processed accoding
     # to SMART Health Cards framework and any constraints created by subclasses
     def strip_fhir_bundle
-      stripped_bundle = @bundle.to_hash
-      if stripped_bundle.key?('entry') && !stripped_bundle['entry'].empty?
-        entries = stripped_bundle['entry']
-        entries, @url_map = redefine_uris(entries)
-        update_elements(entries)
+      return [] unless bundle.entry
+
+      # Using this to dup the original bundle, hopefully it correctly dups everything
+      new_bundle = FHIR::Bundle.new(bundle.to_hash)
```

Comment:
I investigated deep_dup. Deep_dup for an object just calls dup, so it seemed like this would be good enough. It's easy enough to externalize this into a method though.